ddtrace/profile/_periodic.py

```diff
@@ -0,0 +1,120 @@
+# -*- encoding: utf-8 -*-
```

Comment:
Should this be moved to `ddtrace/internal` and replace our existing implementation of `PeriodicThread`?

Reply:
I would argue that localizing this PR as much to just the profile import would be helpful. Refactoring in smaller-sized PRs after this would help moving forward as well.

Reply:
Yes, there are some code there and there that we can mutualize but I'd prefer to do that in a subsequent PR.